src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -2392,12 +2392,16 @@ jmethodID InstanceKlass::update_jmethod_id(jmethodID* jmeths, Method* method, in
 }
 
 // Lookup or create a jmethodID.
-// This code is called by the VMThread and JavaThreads so the
-// locking has to be done very carefully to avoid deadlocks
-// and/or other cache consistency problems.
-//
-jmethodID InstanceKlass::get_jmethod_id(const methodHandle& method_h) {
-  Method* method = method_h();
+static jmethodID* create_jmethod_id_cache(size_t size) {
+  jmethodID* jmeths = NEW_C_HEAP_ARRAY(jmethodID, size+1, mtClass);
+  memset(jmeths, 0, (size+1)*sizeof(jmethodID));
+  // cache size is stored in element[0], other elements offset by one
+  jmeths[0] = (jmethodID)size;
+  return jmeths;
+}
+
+
+jmethodID InstanceKlass::get_jmethod_id(Method* method) {
```

Comment:
Should `method` be `const`?

Reply:
Much of the metadata parameter declarations aren't const-safe.  Adding this one const has a big fall-out leading to some jni calls.  This would be is a good project and we should have it for new code, which some of this is.  I'll see if can add any consts.

Reply:
Thanks for investigating!